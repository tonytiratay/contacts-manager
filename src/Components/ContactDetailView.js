import React, { Component } from 'react';
import SelectTag from './SelectTag';
import style from '../style/contactDetailView';
import PropTypes from 'prop-types';

const isEven = (num) => {
	return num % 2 === 0
};
const grey = "#efefef";

class ContactDetailView extends Component {

	handleEdit(e){
		e.preventDefault();
		let { name, value } = e.target;
		this.props.handleEditUserInfo({ [name]: value });
	}

	tags(){
		let { tags, user } = this.props;
		return <SelectTag options={tags} value={user.tags} addTag={this.addTag.bind(this)} removeTag={this.removeTag.bind(this)}/>;
	}

	addTag(tag){
		let newUser = this.props.user;
		this.props.handleEditUserInfo({ tags: [...newUser.tags, tag] });
	}

	deleteContact(contactId){
		return window.confirm("Êtes vous sur de vouloir effacer ce contact ?") ? this.props.deleteContact(contactId) : false;
	}

	removeTag(tag){
		let tags = this.props.user.tags;
		// Find tag index
		function findTag(elem, tagToFind) {
		  return elem === tagToFind;
		}
		let index = tags.findIndex(findTag.bind(this, tag));
		// Remove tag
		let newTags = [...tags];
		newTags.splice(index, 1);
		this.props.handleEditUserInfo({ tags: newTags });
	}

	render(){
		let { user } = this.props;
		if (user) {
			let { name, avatar, company, email, githubName, id } = this.props.user;
			return (
				<div style={style.container}>
					<div style={style.userContainer}>
						<div style={style.imageContainer}>
							<div style={{ ...style.image, 
								backgroundImage: `url(${avatar})`, 
								border: isEven(this.props.index) ? '2px solid #fff' : '2px solid ' + grey,}}>
							</div>
						</div>
						<div style={style.userInfo}>
							<div>
								<input 
									style={style.inputName} 
									type="text" 
									name="name"
									onChange={this.handleEdit.bind(this)}
									value={name || 'non renseigné'}/>
							</div>
							<div><b style={style.label}>Travail chez: </b>
								<input 
									style={style.inputCompany} 
									type="text" 
									name="company"
									onChange={this.handleEdit.bind(this)}
									value={company || 'non renseigné'}/>
							</div>
							<div><b style={style.label}>Adresse mail: </b>
								<input 
									style={style.inputCompany} 
									type="text" 
									name="email"
									onChange={this.handleEdit.bind(this)}
									value={email || 'non renseigné'}/>
							</div>
							<div><b style={style.label}>Profil Github: </b>
								<input 
									style={style.inputCompany} 
									type="text" 
									name="githubName"
									onChange={this.handleEdit.bind(this)}
									value={githubName || 'non renseigné'}/>
							</div>
						</div>
						<div style={style.delete}>
							<div style={ style.button} onClick={this.deleteContact.bind(this, id)}>Supprimer</div>
						</div>
					</div>
					<div style={style.infos}>
						<div style={ style.formInputs }>
							<div style={ style.formInputContainer }>
								<div><label style={ style.label } htmlFor="tags">Tags associés</label></div>
								{this.tags()}
							</div>
						</div>
					</div>
				</div>
			);
		} else {
			return <div>Pas d'utilisateur à afficher</div>;
		}
	}
}

ContactDetailView.propTypes = {
	user: PropTypes.object,
	tags: PropTypes.array,
	handleEditUserInfo: PropTypes.func,
	deleteContact: PropTypes.func,
}

export default ContactDetailView;