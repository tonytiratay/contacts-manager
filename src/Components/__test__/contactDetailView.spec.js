import React from 'react';
import renderer from 'react-test-renderer';

import ContactDetailView from '../ContactDetailView';

it('Should render correctly', ()=>{

	const user = {
		avatar: 'http://avatarurl.com/pic.jpg',
		name: 'JohnDoe',
		company: 'Leikir',
		tags: ['javascript', 'react'],
	};
	const tags=['javascript, meteor, redux, react, node, python'];

	const tree = renderer
		.create(<ContactDetailView user={user} tags={tags}/>)
		.toJSON();

	expect(tree).toMatchSnapshot();
})