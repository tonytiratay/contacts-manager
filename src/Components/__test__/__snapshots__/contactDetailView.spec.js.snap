// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "padding": 20,
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "justifyItems": "flex-start",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "marginLeft": "10px",
          "marginRight": "10px",
        }
      }
    >
      <div
        style={
          Object {
            "alignSelf": "center",
            "backgroundImage": "url(http://avatarurl.com/pic.jpg)",
            "backgroundSize": "cover",
            "border": "2px solid #efefef",
            "borderRadius": "50%",
            "height": "100px",
            "width": "100px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignSelf": "center",
          "marginLeft": "10px",
        }
      }
    >
      <div>
        <input
          name="name"
          onChange={[Function]}
          style={
            Object {
              "border": "none",
              "color": "#333",
              "fontSize": "30px",
              "fontWeight": "300",
            }
          }
          type="text"
          value="JohnDoe"
        />
      </div>
      <div>
        <b
          style={
            Object {
              "color": "#666",
              "fontSize": 13,
              "marginRight": 5,
            }
          }
        >
          Travail chez: 
        </b>
        <input
          name="company"
          onChange={[Function]}
          style={
            Object {
              "border": "none",
              "fontSize": 14,
              "marginLeft": 10,
            }
          }
          type="text"
          value="Leikir"
        />
      </div>
      <div>
        <b
          style={
            Object {
              "color": "#666",
              "fontSize": 13,
              "marginRight": 5,
            }
          }
        >
          Adresse mail: 
        </b>
        <input
          name="email"
          onChange={[Function]}
          style={
            Object {
              "border": "none",
              "fontSize": 14,
              "marginLeft": 10,
            }
          }
          type="text"
          value="non renseigné"
        />
      </div>
      <div>
        <b
          style={
            Object {
              "color": "#666",
              "fontSize": 13,
              "marginRight": 5,
            }
          }
        >
          Profil Github: 
        </b>
        <input
          name="githubName"
          onChange={[Function]}
          style={
            Object {
              "border": "none",
              "fontSize": 14,
              "marginLeft": 10,
            }
          }
          type="text"
          value="non renseigné"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyItems": "flex-end",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "background": "#ff0000",
            "borderRadius": 10,
            "color": "#fff",
            "cursor": "pointer",
            "flex": 1,
            "height": "17px",
            "margin": "10px",
            "padding": 10,
            "textAlign": "center",
          }
        }
      >
        Supprimer
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "marginTop": 20,
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid #efefef",
          "display": "flex",
          "flexWrap": "wrap",
          "marginBottom": 15,
          "padding": 15,
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
            "flexWrap": "wrap",
            "margin": "10px",
          }
        }
      >
        <div>
          <label
            htmlFor="tags"
            style={
              Object {
                "color": "#666",
                "fontSize": 13,
                "marginRight": 5,
              }
            }
          >
            Tags associés
          </label>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexWrap": "wrap",
            }
          }
        >
          <div
            style={
              Object {
                "border": "1px solid #efefef",
                "flex": 1,
                "margin": 10,
                "minHeight": 170,
                "minWidth": 250,
                "padding": 10,
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "flex": 1,
                  "padding": 5,
                }
              }
            >
              javascript, meteor, redux, react, node, python
            </div>
          </div>
          <div
            style={
              Object {
                "border": "1px solid #efefef",
                "flex": 1,
                "margin": 10,
                "minHeight": 170,
                "minWidth": 250,
                "padding": 10,
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "borderLeft": "3px solid #bdf2ae",
                  "cursor": "pointer",
                  "flex": 1,
                  "padding": 6,
                }
              }
            >
              javascript
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "borderLeft": "3px solid #bdf2ae",
                  "cursor": "pointer",
                  "flex": 1,
                  "padding": 6,
                }
              }
            >
              react
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
